Server/src/index.ts

```diff
@@ -313,18 +345,143 @@ export class CitrineOSServer {
+      const module = new EVDriverModule(
+        this._config,
+        this._cache,
+        this._createSender(),
+        this._createHandler(),
+        this._logger,
+        this._repositoryStore.authorizationRepository,
+        this._repositoryStore.deviceModelRepository,
+        this._repositoryStore.tariffRepository,
+      );
+      this.modules.push(module);
+      this.apis.push(new EVDriverModuleApi(module, this._server, this._logger));
+    }
+
+    if (this._config.modules.monitoring) {
+      const module = new MonitoringModule(
+        this._config,
+        this._cache,
+        this._createSender(),
+        this._createHandler(),
+        this._logger,
+        this._repositoryStore.deviceModelRepository,
+        this._repositoryStore.variableMonitoringRepository,
+      );
+      this.modules.push(module);
+      this.apis.push(
+        new MonitoringModuleApi(module, this._server, this._logger),
+      );
+    }
+
+    if (this._config.modules.reporting) {
+      const module = new ReportingModule(
+        this._config,
+        this._cache,
+        this._createSender(),
+        this._createHandler(),
+        this._logger,
+        this._repositoryStore.deviceModelRepository,
+        this._repositoryStore.securityEventRepository,
+        this._repositoryStore.variableMonitoringRepository,
+      );
+      this.modules.push(module);
+      this.apis.push(
+        new ReportingModuleApi(module, this._server, this._logger),
+      );
+    }
+
+    if (this._config.modules.smartcharging) {
+      const module = new SmartChargingModule(
+        this._config,
+        this._cache,
+        this._createSender(),
+        this._createHandler(),
+        this._logger,
+      );
+      this.modules.push(module);
+      this.apis.push(
+        new SmartChargingModuleApi(module, this._server, this._logger),
+      );
+    }
+
+    if (this._config.modules.transactions) {
+      const module = new TransactionsModule(
+        this._config,
+        this._cache,
+        this._createSender(),
+        this._createHandler(),
+        this._logger,
+        this._repositoryStore.transactionEventRepository,
+        this._repositoryStore.authorizationRepository,
+        this._repositoryStore.deviceModelRepository,
+        this._repositoryStore.tariffRepository,
+      );
+      this.modules.push(module);
+      this.apis.push(
+        new TransactionsModuleApi(module, this._server, this._logger),
+      );
+    }
+
+    // TODO: take actions to make sure module has correct subscriptions and log proof
+    if (this.eventGroup !== EventGroup.All) {
+      this.host = this._config.centralSystem.host as string;
+      this.port = this._config.centralSystem.port as number;
+    }
+  }
+
+  private startOcpiServer(host: string, port: number) {
+    const ocpiServer = new OcpiServer(
+      this._config as OcpiServerConfig,
+      this._cache,
+      this._logger,
+      this.getOcpiModuleConfig(),
+      this._repositoryStore,
```

Comment:
Seemingly this additional argument requires changes in ocpi repo?

Can we please have a PR up for this change in ocpi repo? Not seeing one at the moment.

Reply:
@langiole this is already in `main`: 
```
export class OcpiServer extends KoaServer {
  koa!: Koa;
  readonly serverConfig: OcpiServerConfig;
  readonly cache: ICache;
  readonly logger: Logger<ILogObj>;
  readonly ocpiSequelizeInstance: OcpiSequelizeInstance;
  readonly modules: OcpiModule[] = [];
  readonly repositoryStore: RepositoryStore;
  
  constructor(
    serverConfig: OcpiServerConfig,
    cache: ICache,
    logger: Logger<ILogObj>,
    modulesConfig: OcpiModuleConfig[],
    repositoryStore: RepositoryStore,
  ) {
  ...
  ```
  I think you are talking about these changes right?